### DISCUSSION TITLE
Payload and React Native

### DISCUSSION THREAD

**Discussion by brianFrizzera** [payloadcms/payload#9121] [Q&A]
Hi everyone,

I'm planning the tech stack for a new app with these core requirements:

1. User authentication;

2. Subscription-based paid content;

3. A user space for storing some files;

4. Built with React Native;

5. This same content will be displayed in a web app.

I considered using Payload CMS to manage users and subscriptions, but I’ve hit a roadblock. I can’t figure out how to restrict access to certain app features using Payload's RBAC (Role-Based Access Control). An API might bridge the gap, but I'm uncertain about configuring it to enforce access restrictions effectively.

Does anyone have experience with this? Is it feasible, or should I consider another approach?

Happy to provide more details if needed.

Thanks in advance!

**Comment by brianFrizzera** [comment]
> Hi everyone,
> 
> I'm planning the tech stack for a new app with these core requirements:
> 
> 1. User authentication;
> 
> 
> 2. Subscription-based paid content;
> 
> 
> 3. A user space for storing some files;
> 
> 
> 4. Built with React Native;
> 
> 5. This same content will be displayed in a web app.
> 
> 
> 
> I considered using Payload CMS to manage users and subscriptions, but I’ve hit a roadblock. I can’t figure out how to restrict access to certain app features using Payload's RBAC (Role-Based Access Control). An API might bridge the gap, but I'm uncertain about configuring it to enforce access restrictions effectively.
> 
> Does anyone have experience with this? Is it feasible, or should I consider another approach?
> 
> Happy to provide more details if needed.
> 
> Thanks in advance!

Up

**Comment by vladvlasov256** [comment]
We use Payload CMS API in mobile apps (SwiftUI, iOS and Compose, Android).
We use role-based access. Basically the end users (or apps) have read-only access to the collections. And admins create the content. And super-admins may create admin users. So we have 3 roles.

**Comment by aaronksaunders** [comment]
you need to write custom access rules on the collection objects and then it will work, i am referring to v3, I never tried in v2
